### PR TITLE
Add a warning about the use of gazebo_msgs without Gazebo Classic

### PR DESCRIPTION
**Pull request by j-rivero**
Related to https://github.com/ros-simulation/gazebo_ros_pkgs/issues/1532.

Add a warning for ROS 2 distributions different than humble and iron (current supported ones) that keeps Gazebo users aware of the real goal of the package (being msgs for Gazebo classic).

### CODE REVIEW

**Review by ahcorde — APPROVED** [review]

File: gazebo_msgs/CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,20 @@ cmake_minimum_required(VERSION 3.5)
 
 project(gazebo_msgs)
 
+if(NOT DEFINED IGNORE_GAZEBO_MSGS_WARNING)
+  if(DEFINED ENV{ROS_DISTRO})
+    set(ROS_DISTRO $ENV{ROS_DISTRO})
+    if(NOT (ROS_DISTRO STREQUAL "iron" OR
+            ROS_DISTRO STREQUAL "humble"))
+      message(WARNING "This gazebo_msgs package hosts messages designed initially for Gazebo Classic "
+        "which is not available on ROS 2 Jazzy which is unavailable on ROS 2 Jazzy. "
```
Comment (azeey) [inline gazebo_msgs/CMakeLists.txt]: Is this what you meant?
```suggestion
        "which is not available on ROS 2 Jazzy which is unavailable on Ubuntu Noble. "
```

File: gazebo_msgs/CMakeLists.txt [hunk 2]
```diff
@@ -2,6 +2,20 @@ cmake_minimum_required(VERSION 3.5)
 
 project(gazebo_msgs)
 
+if(NOT DEFINED IGNORE_GAZEBO_MSGS_WARNING)
```
Comment (azeey) [inline gazebo_msgs/CMakeLists.txt]: Most users will not be building this package from source so they will not be seeing this warning. Ideally, we would print the warning when the downstream application is doing a `find_package`.

Comment (azeey) [reply gazebo_msgs/CMakeLists.txt]: Chris suggested using `CONFIG_EXTRAS` with `ament_package` , e.g https://github.com/ros2/rclcpp/blob/rolling/rclcpp_components/CMakeLists.txt#L176

Comment (j-rivero) [reply gazebo_msgs/CMakeLists.txt]: good catch. [7037648](https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1533/commits/70376483630d04361461f4e00e13a1920ac48a99) Still need to test it, closing by today.

Comment (azeey) [reply gazebo_msgs/CMakeLists.txt]: Okay, thanks. I'd be okay with removing this and only doing the `CONFIG_EXTRAS` thing, but I'll leave it up to you.

**Review by azeey — APPROVED** [review]

**Pull request merged**